dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryAPI.java

```diff
@@ -496,4 +499,38 @@ Category findByVariable(final String variable, final User user,
 	 */
 	PaginatedCategories findAll(final CategorySearchCriteria searchCriteria, final User user, boolean respectFrontendRoles)
 			throws DotDataException, DotSecurityException;
+
+	/**
+	 * Find Categories by inodes and calculate its Hierarchy.
+	 *
+	 * For Example if you have the follows categories:
+	 *
+	 *
+	 * | Inode  | Name        |  key        |Parent          |
+	 * |--------|-------------|-------------|----------------|
+	 * | 1      | Top Category| top         | null           |
+	 * | 2      | Child       | child       | Top Category   |
+	 * | 3      | Grand Child | grand_child | Child          |
+	 *
+	 * And you find by inode 3 then you got:

```

Comment:
and you search** by inode 3

Reply:
done https://github.com/dotCMS/core/pull/29142/commits/7673fdbe9ce3848b65d22e4623eb37162652b0fa#diff-cc8d6686856cf74ed3cfdbbf782839a205ade11ffa80218a0a1365fe47ac7c17L515